M2/Macaulay2/packages/Cremona.m2

```diff
@@ -884,13 +963,13 @@ inverseMapInt (RationalMap,Nothing) := o -> (Phi,nothing) -> (
    n := Phi#"dimAmbientTarget"; 
    Sub := map(target Phi,ring Bl,matrix{{(n+1):0_(ambient target Phi)}}|(vars ambient target Phi));
    T := transpose mingens kernel transpose Sub submatrix(jacobian Bl,{0..n},);
-   psi := try map rationalMap(target Phi,source Phi,submatrix(T,{0},)) else error "not able to obtain an inverse rational map";
+   psi := try map rationalMap(target Phi,source Phi,submatrix(T,{0},)) else error "not able to obtain an inverse rational map; make sure the input map is birational";
    psiAll := apply(numRows T,i -> map(target Phi,source Phi,submatrix(T,{i},)));
    if not o.MathMode then return (psi,psiAll);
    if (if n - Phi#"dimTarget" <= Phi#"dimAmbientSource" - Phi#"dimSource" then isInverseMap(map Phi,psi) else isInverseMap(psi,map Phi)) then (
         if o.Verbose then <<certificate; return (psi,psiAll);
    ) else (
-        error "not able to obtain an inverse rational map";
+        error "not able to obtain an inverse rational map; make sure the input map is birational";
```

Comment:
Could the code fail to find the inverse of a birational map?  If so, you might say "not able to obtain an inverse rational map; the map may not be birational".  If not, say "map is not birational".

The problem with the current way of saying it is: what should the user do if he/she has already ensured the map is birational?  It seems to imply that the code would always succeed in finding the inverse, but it's a bit ambiguous.

Reply:
OK, thanks. I correct it immediately.